### PR TITLE
Follow up for complex DOM versions

### PR DESCRIPTION
**Pull request by issackjohn**
This is a follow up for PR #264 plus some additional promised clean up. No changes to performance.

This PR includes the following changes:

- adding comments
- check that JSDOM is present
- Perform prerequisites for building in the complex versions of the TodoMVC in the big-dom-generator BuildComplex utility function and handle them.
- code clean up
- remove dev dependency of jsdom in build script and move it to big-dom-generator BuildComplex Utility function

### CODE REVIEW

**Comment by issackjohn** [comment]
@julienw Could you PTAL, I've only made the changes for the Angular-Complex version. After we've reviewed and approved how we do it for Angular, I can perform the same changes to the rest of the build scripts.

File: resources/todomvc/big-dom-generator/utils/buildComplex.js
Comment (julienw) [inline resources/todomvc/big-dom-generator/utils/buildComplex.js:23]: Thanks for the extra description of the options parameter!

File: resources/todomvc/big-dom-generator/utils/buildComplex.js
Comment (julienw) [inline resources/todomvc/big-dom-generator/utils/buildComplex.js:151]: Thanks for the new comment!

**Review by julienw — COMMENTED** [review]
This looks good to me!
Can you please share a bit more the high-level purpose for this patch? My understanding is that this especially ensures that all dependencies are properly installed in all the related repositories. Can you please confirm?

**Review by julienw — APPROVED** [review]

**Comment by issackjohn** [comment]
When I build the vue workload there are changes to the dist and I'm not sure why. Maybe it was updated but the dist wasn't rebuilt in a previous PR?

**Review by rniwa — APPROVED** [review]

**Comment by issackjohn** [comment]
@julienw, @rniwa  Should I wait for more approvals or can I merge?

**Comment by rniwa** [comment]
I think we technically want Google not being opposed to this change within 10 business days.

**Review by flashdesignory — APPROVED** [review]

File: resources/todomvc/architecture-examples/lit-complex/package.json
Comment (julienw) [inline resources/todomvc/architecture-examples/lit-complex/package.json:15]: I just realized that package-lock.json wasn't regenerated after this change, would it be possible to run `npm i` in all affected projects and commit package-lock.json?

Comment (issackjohn) [reply resources/todomvc/architecture-examples/lit-complex/package.json:15]: https://github.com/WebKit/Speedometer/pull/339

**Pull request merged**
